packages/google_maps_flutter/google_maps_flutter_android/lib/src/google_maps_flutter_android.dart

```diff
@@ -327,6 +327,18 @@ class GoogleMapsFlutterAndroid extends GoogleMapsFlutterPlatform {
     );
   }
 
+  @override
+  Future<void> updateHeatmaps(
+    HeatmapUpdates heatmapUpdates, {
+    required int mapId,
+  }) {
+    assert(heatmapUpdates != null);
+    return _channel(mapId).invokeMethod<void>(
+      'heatmaps#update',
+      heatmapUpdates.toJson(),
```

Comment:
I don't follow; addressing this would mean that instead of using `heatmapUpdates.toJson()` you would use, e.g. `_methodChannelRepresentationForHeatmapUpdates(heatmapUpdates)`, where that's a new method that creates exactly the same structure that you are expecting on the native side (which presumably would happen, at the moment, to be the same implementation as `HeatmapUpdates.toJson` currently has). Why would that require changes to anything else?

Reply:
My thoughts on this:
- The base `MapsObject` class has a `toJson` method which means the `Heatmap` class must have one. Is it okay to just throw an `UnimplementedError` here until the rest of the code is refactored for this?
- I understand the desire to have a platform-specific implementation of everything, but this code is the same on all platforms. Making it platform-specific is just a copy/paste job and makes updating things for all platforms later a much bigger task.

Reply:
> * The base `MapsObject` class has a `toJson` method which means the `Heatmap` class must have one. Is it okay to just throw an `UnimplementedError` here until the rest of the code is refactored for this?

I forgot about the base class requiring `toJson`. It's fine to leave the implementation you've added for it then, it'll just be unused within the plugin.

> * I understand the desire to have a platform-specific implementation of everything, but this code is the same on all platforms. Making it platform-specific is just a copy/paste job and makes updating things for all platforms later a much bigger task.

No, it makes future updates easier. The serialization here is one half of a cross-language communication channel, the other half of which is in platform specific native code. Under the current structure, making any changes—even if it's only for a singe platform—requires coordinating changes across *three different packages*, several of which cannot be changed in the same PR. That's a very fragile setup, which exists only because originally of this code was in a single package, and it wasn't refactored when that package was split apart.

Having the serialization and deserialization in the same package means any change to the platform channel is entirely contained within one package, instead of needlessly entangled with other packages.

Reply:
That makes sense. I'll get on it.

Reply:
Let me know what you think about my implementation for this